pkg/controller/controlplane/utils.go

```diff
@@ -33,8 +36,42 @@ func DNSNamesForService(name, namespace string) []string {
 	}
 }
 
-// MergeSecretMaps merges the 2 given secret maps.
-func MergeSecretMaps(a, b map[string]*corev1.Secret) map[string]*corev1.Secret {
+// ComputeChecksums computes and returns the checksums for the given secrets and configmaps,
+// as well as the secrets and configmaps with the given names that are fetched from the cluster.
+func ComputeChecksums(
```

Comment:
I'd rather still have the old `Merge...` methods and the old `ComputeChecksums` method. This one has too many responsibilities at once. Also, `Get`ting secrets and configmaps one by one by name is not really well performing and hides away the fact that one should rather apply smarter labels to all relevant secrets in order to fetch them in one request.
So my suggestion:
* Keep the old methods
* Apply better labels to relevant secrets
* Maybe create a method `AddSecretsToSecretsMap(secrets map[string]*corev1.Secret, newSecrets []*corev1.Secret)` (same for `ConfigMap`) that simplify building up the secrets and configmap map.

Reply:
* The new method is exactly what `Actuators` would need to use in order to correctly build their checksums. I do believe a single method is very useful, otherwise its code will be copied for every provider. 
* In practice, it only reads 1 secret (cloudprovider) and 1 config map (cloud-provider-config) from the cluster, so performance considerations are not relevant. Adding labels etc. would only complicate code and would be a "premature optimization" at best. 
* The old methods are still there, I could just make them public if you prefer - but that would be increasing API surface prematurely.

Reply:
The new method is what exactly your `controlplane` actuators need to do + it is heavily overloaded and non-performant. Testing all the edge cases of this single function is way too heavy duty. Also, if we split it up now, we can re-use this function easily in the future.

And even if it only reads one secret, if we in the future decide to add more of them, then we suddenly get performance issues due to the multiple `Get` requests - why build in performance pits if we can easily avoid them??